From 87343cd4bdf5fa1433a809f5268af1e0d832ad83 Mon Sep 17 00:00:00 2001
From: Josh Guilfoyle <Josh.Guilfoyle@T-Mobile.com>
Date: Wed, 26 Jan 2011 23:28:43 -0800
Subject: [PATCH 08/51] OMS7-N: Disable Zygote preloaded drawables [8/11]

With a theme applied, most of these preloaded drawables go unused.  Any
assets the theme has redirected will need to be loaded with each app
process regardless.  Worse, preloads make it impossible to do asset
redirection for constituent parts of a preloaded drawable (for instance,
individual states of a StateListDrawable cannot be redirected).

Some day it might be nice to revisit this and see if there's a way to
reintroduce the drawable cache in a way that can be altered at runtime
without significant complexity or runtime penalty.

Change-Id: I253b1a22482ac664c196533a4c2fcd88ae84b996
---
 core/java/com/android/internal/os/ZygoteInit.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/java/com/android/internal/os/ZygoteInit.java b/core/java/com/android/internal/os/ZygoteInit.java
index cc71d37ea1c..8278b90de0d 100644
--- a/core/java/com/android/internal/os/ZygoteInit.java
+++ b/core/java/com/android/internal/os/ZygoteInit.java
@@ -109,7 +109,7 @@ public class ZygoteInit {
     private static final String PRELOADED_CLASSES = "/system/etc/preloaded-classes";
 
     /** Controls whether we should preload resources during zygote init. */
-    public static final boolean PRELOAD_RESOURCES = true;
+    public static final boolean PRELOAD_RESOURCES = false;
 
     /**
      * Registers a server socket for zygote command connections
@@ -440,6 +440,8 @@ public class ZygoteInit {
                     Log.i(TAG, "...preloaded " + N + " resource in "
                             + (SystemClock.uptimeMillis() - startTime) + "ms.");
                 }
+            } else {
+                Log.i(TAG, "Preload resources disabled, skipped.");
             }
             mResources.finishPreloading();
         } catch (RuntimeException e) {
-- 
2.11.1

